### PR TITLE
fix(GUI): ensure settings are loaded before bootstrapping application

### PR DESCRIPTION
**Pull request by jviotti**
We load localStorage settings into the Redux store in an asynchronous
way. This means that user settings might not be loaded by the time the
application starts, resulting in Mixpanel sending a few tracking events
before Etcher realises that the user opted out from anonoymous analytics
and error reporting.

In order to fix that, we remove `ng-app` and we manually bootstrap the
Angular.js application *after* the local settings are loaded.

Change-Type: patch
Changelog-Entry: Don't send initial Mixpanel events before "Anonymous Tracking" settings are loaded.
Fixes: https://github.com/resin-io/etcher/issues/1772
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

File: lib/gui/app.js [hunk 1]
```diff
@@ -102,77 +102,75 @@ app.run(() => {
...
+  const updatesEnabled = settings.get('updatesEnabled')
+
+  if (!shouldCheckForUpdates || !updatesEnabled) {
+    analytics.logEvent('Not checking for updates', {
+      shouldCheckForUpdates,
+      updatesEnabled,
+      stable: isStableRelease
     })
 
-    const isStableRelease = release.isStableRelease(currentVersion)
```
Comment (lurch) [inline lib/gui/app.js]: Sometimes the diffs that Github produces are so painful to decipher :-(

File: lib/gui/app.js [hunk 2]
```diff
@@ -102,77 +102,75 @@ app.run(() => {
     version: currentVersion
   })
 
-  settings.load().then(() => {
```
Comment (lurch) [inline lib/gui/app.js]: Just out of curiosity, would moving the `analytics.logEvent('Application start'` line below the `settings.load().then(` line also have fixed the problem? Or does it go deeper than that?

Comment (jviotti) [reply lib/gui/app.js]: Yeah, that would have fixed the issue. This PR changes the whole Angular.js bootstrapping method so we can be sure everything will be ready before we execute anything, protecting us from forgetting to move analytic events call below a certain other instruction, etc.

**Comment by lurch** [comment]
This _partially_ fixes #1772 - #1902 was the other half of the fix.

**Pull request merged**